src/rgw/rgw_sync_log_trim.cc

```diff
@@ -347,7 +347,7 @@ int take_min_status(CephContext *cct, Iter first, Iter last,
                     std::vector<std::string> *status)
 {
   status->clear();
-  boost::optional<size_t> num_shards;
+  auto num_shards = boost::make_optional(false, (size_t)(0));
```

Comment:
First of all, this change makes the logic wrong. Here `num_shards` is supposed to make  `if (!num_shards) {` to true and then take the initial value from `peer_shards`. So do not initialize it at L350.

Also this warning seems to be a gcc bug: https://svn.boost.org/trac10/ticket/11169.